### PR TITLE
Soulstone Fix

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Should now properly pull people back before attempting to offer the body

### CODE REVIEW

**Comment by KorPhaeron** [comment]
I thought of a better way to do this that saves some copypaste and a var whoops

**Comment by Zilenan91** [comment]
Does this fix people being completely locked out of the game if they get turned into a construct?

File: code/game/gamemodes/wizard/soulstone.dm [hunk 1]
```diff
@@ -238,24 +238,31 @@
 
 
 /obj/item/device/soulstone/proc/getCultGhost(obj/item/device/soulstone/C, mob/living/carbon/human/T, mob/U)
+	var/success = 0
+	var/client/chosen_ghost
 
-	var/list/consenting_candidates = pollCandidates("Would you like to play as a Shade?", be_special_flag = ROLE_CULTIST, poll_time = 100)
+	for(var/mob/dead/observer/ghost in player_list) //We put them back in their body
+		if(T.real_name == ghost.real_name && ghost.client)
```
Comment (phil235) [inline code/game/gamemodes/wizard/soulstone.dm]: Using names could cause issues if two mobs have the same name, I think. Maybe use `if(T == ghost.mind.current` like the reenter_corpse() proc does.

Comment (KorPhaeron) [inline code/game/gamemodes/wizard/soulstone.dm]: What if I wanna catch the people who ghost while alive?

**Comment by KorPhaeron** [comment]
@AnturK 

Please add this to your PR since they'll just conflict otherwise

**Comment by duncathan** [comment]
Oops I missed this

**Comment by tkdrg** [comment]
@KorPhaeron conflicts

**Comment by KorPhaeron** [comment]
@duncathan 

traitor

**Pull request closed without merging**